types/react-instantsearch/connectors/index.d.ts

```diff
@@ -0,0 +1,206 @@
+import { AlgoliaError, BasicDoc, ConnectedComponentType, SearchResults, SearchState, Hit } from '../types';
+
+export function connectConfigure(...args: any[]): any;
+
+export interface NESW {
+  northEast: { lat: number, lng: number };
+  southWest: { lat: number, lng: number };
+}
+
+export interface GeoSearchExposed {
+  defaultRefinement?: NESW;
+}
+export interface GeoSearchProvided<THit> {
+  /** a function to toggle the refinement */
+  refine?: (refinement: NESW) => void;
+  /** a function to generate a URL for the corresponding search state */
+  createURL?: (...args: any[]) => any;
+  /** the records that matched the search */
+  hits: THit[];
+  /** true if the current refinement is set with the map bounds */
+  isRefinedWithMap?: boolean;
+  /** the refinement currently applied */
+  currentRefinement?: NESW;
+  /** the position of the search */
+  position?: { lat: number, lng: number };
+}
+/**
+ * The GeoSearch connector provides the logic to build a widget that will display the results on a map.
+ * It also provides a way to search for results based on their position. The connector provides function to manage the
+ * search experience (search on map interaction).
+ *
+ * https://community.algolia.com/react-instantsearch/connectors/connectGeoSearch.html
+ */
+export function connectGeoSearch<TProps extends GeoSearchProvided<THit>, THit>(ctor: React.ComponentType<TProps>): ConnectedComponentType<TProps, GeoSearchProvided<THit>, GeoSearchExposed>;
+export function connectCurrentRefinements(...args: any[]): any;
+export function connectHierarchicalMenu(...args: any[]): any;
+export function connectHighlight(...args: any[]): any;
+
+/**
+ * connectHits connector provides the logic to create connected components that will render the results retrieved from Algolia.
+ * To configure the number of hits retrieved, use HitsPerPage widget, connectHitsPerPage connector or pass the hitsPerPage prop to a Configure widget.
+ * Warning: you will need to use the objectID property available on every hit as a key when iterating over them. This will ensure you have the best possible UI experience especially on slow networks.
+ *
+ * https://community.algolia.com/react-instantsearch/connectors/connectHits.html
+ */
+export function connectHits<TProps extends { hits: THit[]}, THit>(ctor: React.ComponentType<TProps>): ConnectedComponentType<TProps, {hits?: THit[]}>;
+
+export function connectAutoComplete(...args: any[]): any;
```

Comment:
These functions need to be in another package called react-instantsearch-core and to be rexported here. You want to have hierarchy that's similar in js files. (I think you need to do for all the files added here)